crawler4j-core/src/main/java/edu/uci/ics/crawler4j/parser/Parser.java

```diff
@@ -67,80 +66,75 @@ public Parser(CrawlConfig config, BasicURLNormalizer normalizer, HtmlParser html
-                    }
-                } else {
-                    parseData.setHtml(Constants.EMPTY_HTML_TAGS);
-                }
-                page.setParseData(parseData);
-                if (parseData.getHtml() == null) {
-                    throw new ParseException();
-                }
-                parseData.setOutgoingUrls(net.extractUrls(parseData.getHtml()));
-            } else {
-                throw new NotAllowedContentException();
-            }
-        } else if (Util.hasCssTextContent(page.getContentType())) { // text/css
-            try {
-                CssParseData parseData = createCssParseData();
-                if (page.getContentCharset() == null) {
-                    parseData.setTextContent(new String(page.getContentData(), StandardCharsets.UTF_8));
-                } else {
-                    parseData.setTextContent(
-                            new String(page.getContentData(), page.getContentCharset()));
-                }
-                parseData.setOutgoingUrls(page.getWebURL());
-                page.setParseData(parseData);
-            } catch (Exception e) {
-                logger.error("{}, while parsing css: {}", e.getMessage(), page.getWebURL().getURL());
-                throw new ParseException();
-            }
-        } else if (Util.hasPlainTextContent(page.getContentType())) { // plain Text
-            try {
-                TextParseData parseData = createTextParseData();
-                if (page.getContentCharset() == null) {
-                    parseData.setTextContent(new String(page.getContentData(), StandardCharsets.UTF_8));
-                } else {
-                    parseData.setTextContent(
-                            new String(page.getContentData(), page.getContentCharset()));
-                }
-                parseData.setOutgoingUrls(net.extractUrls(parseData.getTextContent()));
-                page.setParseData(parseData);
-            } catch (Exception e) {
-                logger.error("{}, while parsing: {}", e.getMessage(), page.getWebURL().getURL());
-                throw new ParseException(e);
-            }
-        } else { // isHTML
-
-            HtmlParseData parsedData = createHtmlParseData(page, contextURL);
-
-            if (page.getContentCharset() == null) {
-                page.setContentCharset(parsedData.getContentCharset());
-            }
-
-            if(config.isLanguageDetection()) {
-                // Please note that identifying language takes less than 10 milliseconds
-                page.setLanguage(languageDetector.detect(parsedData.getText()));
-            } else {
-                page.setLanguage("");
-            }
-
-            page.setParseData(parsedData);
-        }
-    }
-
+		/**
+		 * @throws NotAllowedContentException thrown to enforce not handling the fetched content,
+		 * other exceptions are considered parse exceptions.
+		 */
+		public void parse(Page page, String contextURL)
```

Comment:
see above. Some sort of heavy reformatting mixed with changes. Can you give some pointers?

Reply:
See above.